File: 121. Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        buy = 0
        sell = 1
        profit = 0

        while sell < len(prices):
            # Check profit
            if prices[sell] > prices[buy]:
                profit = max(prices[sell] - prices[buy], profit)
            else:
                buy = sell

            sell += 1

        return profit
            
"""
Time complexity: O(n)
Space complexity: O(1)
"""

File: 121. Best Time to Buy and Sell Stock.c
int maxProfit(int* prices, int pricesSize) {
    // Edge case: only 1 item in prices
    if (pricesSize == 1)
        return 0;

    int max_profit = 0;
    int buy = 0;
    int sell = 1;

    while (sell < pricesSize) {
        if (prices[sell] > prices[buy]) {
            if (prices[sell] - prices[buy] > max_profit) {
                max_profit = prices[sell] - prices[buy];
            }
        } else {
            buy = sell;
        }
        sell++;
    }
    return max_profit;
}